torch/_inductor/utils.py

```diff
@@ -79,6 +79,31 @@ def ceildiv(numer: int, denom: int):
     return -(numer // -denom)
 
 
+def convert_shape_to_inductor(lst: List[Union[int, torch.SymInt]]) -> List[sympy.Expr]:
+    """
+    Gets the shape and stride of a tensor. For non-symbolic tensors, this is
+    trivial. But for symbolic tensors, we need to map from SymIntNode into
+    sympy.Expr.
+    """
+    return [
+        i.node.expr if isinstance(i, torch.SymInt) else sympy.Integer(i) for i in lst
+    ]
+
+
+def convert_shape_to_symint(
+    lst: List[Union[int, sympy.Expr]]
+) -> List[Union[torch.SymInt]]:
+    """
+    Takes a list of shapes from Inductor and converts them into symints (or just
+    ints if all shapes are static).
+    """
+    if all(isinstance(i, int) for i in lst):
+        return lst
+    if all(isinstance(i, sympy.Integer) for i in lst):
+        return [int(i) for i in lst]
+    return [V.graph.sizevars.shape_env.create_symintnode(i) for i in lst]
```

Comment:
At least two bugs here:
- `V` is undefined (just need to import a module)
- `Union` in the result type should be applied to multiple types OR shouldn't be used.

Not sure why the linter didn't catch this. But VS Code makes it clear where the errors are.

This broke some tests on [my dyn-shapes PR](https://github.com/pytorch/pytorch/pull/92667). I'll fix that in that stack since that's a good repro. No need to revert this one, I think.

cc @ezyang @Chillee

Reply:
Hmm, even after doing the quick fixes mentioned above, on my branch with tests, a bunch of tests start failing after these changes.

Here's what I'm going to do: I'll push a commit that reverts this to my stack, then undo the revert after the tests are enabled, so we could see how much stuff is affected on CI.

Reply:
cc @Chillee

Reply:
should be fixed by https://github.com/pytorch/pytorch/pull/92774